### ISSUE TITLE
Cisco Genie processing failing with Netmiko

### ISSUE THREAD

**Issue by vicaceve** [nornir-automation/nornir#676]
I am running into a similar issue as found here https://github.com/nornir-automation/nornir/issues/440. I first ran into this when using nornir, but while I was troubleshooting and making my code simpler I noticed I see this issue with just using netmiko and genie. Not sure if we can tackle this here as the same components are at play and the symptoms are the same - some device output gets parsed correctly while the other does not.

Simplified code:

```python
from netmiko import ConnectHandler
from pprint import pprint

device1 = {
       "device_type": "cisco_xe",
       "host": "x.x.x.x",
       "username": "admin",
       "password": "admin",
   }

 device2 = {
     "device_type": "cisco_xe",
     "host": "x.x.x.x",
     "username": "admin",
     "password": "admin",
 }

 devices = [device1, device2]

 for device in devices:
     with ConnectHandler(**device) as net_connect:
         output = net_connect.send_command(
             "show version",
             use_genie=True
         )
         
         pprint(output)
``` 

Does not matter If I flip the order of the devices in the list, the result always ends up being the same - the first device output never gets parsed, while the second device output does. If I run the script against just 1 device, it also does not get parsed. 

I have been trying to troubleshoot this on my own but haven't gotten anywhere yet. I was using python 3.7.5 and upgraded to 3.8.7 and still showing the same behavior. 

```python
('Cisco IOS XE Software, Version 17.03.03\n'
 'Cisco IOS Software [Amsterdam], Catalyst L3 Switch Software (CAT9K_IOSXE), '
 'Version 17.3.3, RELEASE SOFTWARE (fc7)\n'
 'Technical Support: http://www.cisco.com/techsupport\n'
 'Copyright (c) 1986-2021 by Cisco Systems, Inc.\n'
 'Compiled Thu 04-Mar-21 12:32 by mcpre\n'
 '\n'
 '\n'
 'Cisco IOS-XE software, Copyright (c) 2005-2021 by cisco Systems, Inc.\n'
 'All rights reserved.  Certain components of Cisco IOS-XE software are\n'
 'licensed under the GNU General Public License ("GPL") Version 2.0.  The\n'
 'software code licensed under GPL Version 2.0 is free software that comes\n'
 'with ABSOLUTELY NO WARRANTY.  You can redistribute and/or modify such\n'
 'GPL code under the terms of GPL Version 2.0.  For more details, see the\n'
 'documentation or "License Notice" file accompanying the IOS-XE software,\n'
 'or the applicable URL provided on the flyer accompanying the IOS-XE\n'
 'software.\n'
 '\n'
 '\n'
 'ROM: IOS-XE ROMMON\n'
 'BOOTLDR: System Bootstrap, Version 16.12.1r[FC1], RELEASE SOFTWARE (P)\n'
 '\n'
 'SW1 uptime is 5 weeks, 5 days, 23 hours, 8 minutes\n'
 'Uptime for this control processor is 5 weeks, 5 days, 23 hours, 9 minutes\n'
 'System returned to ROM by Reload Command at 20:49:53 UTC Wed Mar 10 2021\n'
 'System restarted at 20:52:09 UTC Wed Mar 10 2021\n'
 'System image file is "flash:17.03.03/packages.conf"\n'
 'Last reload reason: Reload Command\n'
 '\n'
 '\n'
 '\n'
 'This product contains cryptographic features and is subject to United\n'
 'States and local country laws governing import, export, transfer and\n'
 'use. Delivery of Cisco cryptographic products does not imply\n'
 'third-party authority to import, export, distribute or use encryption.\n'
 'Importers, exporters, distributors and users are responsible for\n'
 'compliance with U.S. and local country laws. By using this product you\n'
 'agree to comply with applicable laws and regulations. If you are unable\n'
 'to comply with U.S. and local laws, return this product immediately.\n'
 '\n'
 'A summary of U.S. laws governing Cisco cryptographic products may be found '
 'at:\n'
 'http://www.cisco.com/wwl/export/crypto/tool/stqrg.html\n'
 '\n'
 'If you require further assistance please contact us by sending email to\n'
 'export@cisco.com.\n'
 '\n'
 '\n'
 'Technology Package License Information: \n'
 '\n'
 '------------------------------------------------------------------------------\n'
 'Technology-package                                     Technology-package\n'
 'Current                        Type                       Next reboot  \n'
 '------------------------------------------------------------------------------\n'
 'network-advantage   \tSmart License                 \t network-advantage   \n'
 'dna-advantage       \tSubscription Smart License    \t '
 'dna-advantage                 \n'
 'AIR License Level: AIR DNA Advantage\n'
 'Next reload AIR license Level: AIR DNA Advantage\n'
 '\n'
 '\n'
 'Smart Licensing Status: Registration Not Applicable/Not Applicable\n'
 '\n'
 'cisco C9500-40X (X86) processor with 1331521K/6147K bytes of memory.\n'
 'Processor board ID FCW2319F0GS\n'
 '28 Virtual Ethernet interfaces\n'
 '48 Ten Gigabit Ethernet interfaces\n'
 '2 Forty Gigabit Ethernet interfaces\n'
 '2048K bytes of non-volatile configuration memory.\n'
 '16777216K bytes of physical memory.\n'
 '1638400K bytes of Crash Files at crashinfo:.\n'
 '11264000K bytes of Flash at flash:.\n'
 '\n'
 'Base Ethernet MAC Address          : 78:02:b1:3f:5e:00\n'
 'Motherboard Assembly Number        : 73-18140-03\n'
 'Motherboard Serial Number          : FOC23182H89\n'
 'Model Revision Number              : D0\n'
 'Motherboard Revision Number        : B0\n'
 'Model Number                       : C9500-40X\n'
 'System Serial Number               : FCW2319F0GS\n'
 'CLEI Code Number                   : \n'
 '\n'
 '\n'
 'Switch Ports Model              SW Version        SW Image              '
 'Mode   \n'
 '------ ----- -----              ----------        ----------            '
 '----   \n'
 '*    1 50    C9500-40X          17.03.03          CAT9K_IOSXE           '
 'INSTALL\n'
 '\n'
 '\n'
 'Configuration register is 0x102\n')
 
{'version': {'bootldr': 'System Bootstrap, Version 16.12.1r, RELEASE SOFTWARE '
                        '(P)',
             'chassis': 'C9500-40X',
             'chassis_sn': 'FJB2319E01T',
             'compiled_by': 'mcpre',
             'compiled_date': 'Thu 04-Mar-21 12:32',
             'curr_config_register': '0x102',
             'disks': {'crashinfo:.': {'disk_size': '1638400',
                                       'type_of_disk': 'Crash Files'},
                       'flash:.': {'disk_size': '11264000',
                                   'type_of_disk': 'Flash'}},
             'hostname': 'SW2',
             'image_id': 'CAT9K_IOSXE',
             'image_type': 'production image',
             'label': 'RELEASE SOFTWARE (fc7)',
             'last_reload_reason': 'Reload Command',
             'license_level': 'Next reload AIR lic',
             'license_type': 'ense Level: AIR DNA Advantag',
             'main_mem': '1331521',
             'mem_size': {'non-volatile configuration': '2048',
                          'physical': '16777216'},
             'next_reload_license_level': 'e',
             'number_of_intfs': {'Forty Gigabit Ethernet': '2',
                                 'Ten Gigabit Ethernet': '48',
                                 'Virtual Ethernet': '30'},
             'os': 'IOS-XE',
             'platform': 'Catalyst L3 Switch',
             'processor_type': 'X86',
             'returned_to_rom_at': '13:41:17 UTC Wed Mar 10 2021',
             'returned_to_rom_by': 'Reload Command',
             'rom': 'IOS-XE ROMMON',
             'rtr_type': 'C9500-40X',
             'switch_num': {'1': {'active': True,
                                  'mac_address': 'ac:f5:e6:14:33:80',
                                  'mb_assembly_num': '73-18140-03',
                                  'mb_rev_num': 'B0',
                                  'mb_sn': 'FJZ23180RWQ',
                                  'mode': 'INSTALL',
                                  'model': 'C9500-40X',
                                  'model_num': 'C9500-40X',
                                  'model_rev_num': 'D0',
                                  'ports': '50',
                                  'sw_image': 'CAT9K_IOSXE',
                                  'sw_ver': '17.03.03',
                                  'system_sn': 'FJB2319E01T',
                                  'uptime': '5 weeks, 6 days, 6 hours, 18 '
                                            'minutes'}},
             'system_image': 'flash:17.03.03/packages.conf',
             'system_restarted_at': '13:43:37 UTC Wed Mar 10 2021',
             'uptime': '5 weeks, 6 days, 6 hours, 17 minutes',
             'uptime_this_cp': '5 weeks, 6 days, 6 hours, 18 minutes',
             'version': '17.3.3',
             'version_short': '17.3',
             'xe_version': '17.03.03'}}
```

```
❯ pip freeze | grep netmiko
netmiko==3.3.3
nornir-netmiko==0.1.1

❯ pip freeze | grep genie
genie==21.2.2
genie.libs.clean==21.2
genie.libs.conf==21.2
genie.libs.filetransferutils==21.2
genie.libs.health==21.2
genie.libs.ops==21.2
genie.libs.parser==21.2.7
genie.libs.sdk==21.2

❯ python -V
Python 3.8.7
```

Is there something wrong with my code or is there an underlying issue somewhere? Any help or clues would be greatly appreciated!

**Comment by ktbyers** [comment]
The code posted above your `pprint` is outside of your for loop and your indentation of the pprint is wrong? I assume that was just a mistake posting the code to GitHub. Can you correct that and make it match the code you are using to test with?


Kirk

**Comment by vicaceve** [comment]
Hey Kirk,
Ya, just messed up the formatting when I copied it over. Fixed now. 

Thanks,
Vic

**Comment by vicaceve** [comment]
For what it's worth, just tested the exact code on Cisco's DevNET Sandbox and it works:

```
(py3venv) [developer@devbox py3venv]$ python example.py 
{'version': {'chassis': 'CSR1000V',
             'chassis_sn': '99NRG2LPAS1',
             'compiled_by': 'mcpre',
             'compiled_date': 'Tue 28-May-19 12:45',
             'curr_config_register': '0x2102',
             'disks': {'bootflash:.': {'disk_size': '16162815',
                                       'type_of_disk': 'virtual hard disk'},
                       'webui:.': {'disk_size': '0',
                                   'type_of_disk': 'WebUI ODM Files'}},
             'hostname': 'dist-rtr01',
             'image_id': 'X86_64_LINUX_IOSD-UNIVERSALK9-M',
             'image_type': 'production image',
             'last_reload_reason': 'Unknown reason',
             'license_level': 'ax',
             'license_type': 'N/A(Smart License Enabled)',
             'main_mem': '1201684',
             'mem_size': {'non-volatile configuration': '32768',
                          'physical': '3018504'},
             'next_reload_license_level': 'ax',
             'number_of_intfs': {'Gigabit Ethernet': '6'},
             'os': 'IOS-XE',
             'platform': 'Virtual XE',
             'processor_type': 'VXE',
             'returned_to_rom_by': 'reload',
             'rom': 'IOS-XE ROMMON',
             'rtr_type': 'CSR1000V',
             'system_image': 'bootflash:packages.conf',
             'uptime': '26 minutes',
             'uptime_this_cp': '28 minutes',
             'version': '16.11.1b',
             'version_short': '16.11'}}
{'version': {'chassis': 'CSR1000V',
             'chassis_sn': '9Z54AH3PRUV',
             'compiled_by': 'mcpre',
             'compiled_date': 'Tue 28-May-19 12:45',
             'curr_config_register': '0x2102',
             'disks': {'bootflash:.': {'disk_size': '16162815',
                                       'type_of_disk': 'virtual hard disk'},
                       'webui:.': {'disk_size': '0',
                                   'type_of_disk': 'WebUI ODM Files'}},
             'hostname': 'dist-rtr02',
             'image_id': 'X86_64_LINUX_IOSD-UNIVERSALK9-M',
             'image_type': 'production image',
             'last_reload_reason': 'Unknown reason',
             'license_level': 'ax',
             'license_type': 'N/A(Smart License Enabled)',
             'main_mem': '1201684',
             'mem_size': {'non-volatile configuration': '32768',
                          'physical': '3018504'},
             'next_reload_license_level': 'ax',
             'number_of_intfs': {'Gigabit Ethernet': '6'},
             'os': 'IOS-XE',
             'platform': 'Virtual XE',
             'processor_type': 'VXE',
             'returned_to_rom_by': 'reload',
             'rom': 'IOS-XE ROMMON',
             'rtr_type': 'CSR1000V',
             'system_image': 'bootflash:packages.conf',
             'uptime': '26 minutes',
             'uptime_this_cp': '28 minutes',
             'version': '16.11.1b',
             'version_short': '16.11'}}

Their Environment:

```
(py3venv) [developer@devbox py3venv]$ python -V
Python 3.6.8

(py3venv) [developer@devbox py3venv]$ pip freeze | grep netmiko
netmiko==3.3.0

(py3venv) [developer@devbox py3venv]$ pip freeze | grep genie
genie==20.8
genie.libs.clean==20.8.1
genie.libs.conf==20.8
genie.libs.filetransferutils==20.8
genie.libs.health==20.8
genie.libs.ops==20.8
genie.libs.parser==20.8
genie.libs.robot==20.8
genie.libs.sdk==20.8.2
genie.telemetry==20.8
genie.trafficgen==20.8
```

```

**Comment by vicaceve** [comment]
Problem solved! I upgraded genie to 21.2.2 which is what I have in my setup and I saw the same problem. Upgraded genie to 20.1.3 and the problem was fixed. So seems like something is off with genie 21.2.2
